### ISSUE TITLE
[Feature Request] lammps_fix_external_set_virial_global python interface

### ISSUE THREAD

**Issue by amirhajibabaei** [lammps/lammps#2834]
**Summary**

LAMMPS coupling with other code is possible by "fix external" mechanism.
Using C library API it is possible to  virial (stress tensor) which is needed for NPT simulations.
But I couldn't find similar functionality using python API. 

**Detailed Description**

The possibility for setting virial in addition to external force is requested.
See the pseudo code below.

```python
from lammps import lammps

def callback(caller, ntimestep, nlocal, tag, pos, fext):
      # set fext
      fext[:] = ...

      # requested functions:
      caller.set_fix_external_energy_global(...)   #  <- requested feature
      caller.set_fix_external_virial_global(...)      # <- requested feature

lmp = lammps() 
# ...
lmp.command("fix 1 all external pf/callback 1 1")
lmp.set_fix_external_callback("1", callback)
# ...
```

**Further Information, Files, and Links**

file:
lammps/python/lammps/core.py

lines:
299    self.lib.lammps_fix_external_set_energy_global = ...
300    self.lib.lammps_fix_external_set_virial_global = ...

**Comment by akohlmey** [comment]
The requested features are now implemented in the pull request #2837 and should be included in the next patch release of LAMMPS.

If you want to try it out right away, you can access the corresponding feature branch here: https://github.com/akohlmey/lammps/tree/fix-external-python-support.
The python interface for setting the per-atom values is still pending.

If you run into any issues or notice inconsistencies in the documentation, please provide comments on the pull request.

**Comment by amirhajibabaei** [comment]
@akohlmey 
Your prompt fix is greatly appreciated. 
I tried a few short runs and it worked without any errors. 
I will try longer simulations for numerical consistency check, etc.
If there were any issues I will comment on the pull request. 
Therefore I will close this issue here.
